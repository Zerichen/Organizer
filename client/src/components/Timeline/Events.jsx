import PropTypes from 'prop-types'; //strict proptype checking
import React from 'react';
import EventDot from './EventDot';

//list of eventdots
class Events extends React.Component {
    render() {
        const props = this.props;

        //a list to keep due dates extracted from the
        //object passed in
        let eventList = [];
        for(const item in props.eventArray) {
            if(props.eventArray.hasOwnProperty(item)) {
                eventList.push(item);
            }
        }
        console.log(eventList);
        
        return (
            <span>
            <i className="arrow left"></i>
            <ol 
                className='eventsList'
                style={{
                    listStyle: 'none'
                }}
            >
                {eventList.map((date, index) => 
                //provide a list of hws of a date
                    <EventDot
                        key={index}
                        label={date}
                        position={(1+index)*150 + 200 + 'px'}
                        hwArray = {props.eventArray[date]}
                    />
                )}
            </ol>
            <i className="arrow right"></i>
            </span>
        );
    }
}

//requires an array of events
Events.propTypes = {
    eventArray: PropTypes.object.isRequired,
}

export default Events;