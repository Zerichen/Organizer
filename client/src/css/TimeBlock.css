#cardlook {
  margin-top: 10px;
  margin-left: 32%;
  width: 60%;
}

#tcomplete{
  margin-top: 10px;
  margin-left: 450px;
  width: 680px;
}

.custom {
  border: #57068C;
  background-color: #57068C;
}

.TaskName .TaskTime{
  text-align: center;
}