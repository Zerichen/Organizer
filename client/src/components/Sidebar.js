import React from 'react';
import { Nav } from 'react-bootstrap';
import { BrowserRouter as Router, Route} from "react-router-dom";
import '../css/sidebar.css';

class SideBar extends React.Component {
   render() {
    return (
      <Router>
        <Nav id="sidebar" variant="pills" className="flex-column" >
          <SidebarLink activeOnlyWhenExact={true} to="/" label="User"/>
          <SidebarLink to="/schedules" label="Schedules"/>
          <SidebarLink to="/tags" label="Tags"/>
          <SidebarLink to="/analysis" label="Analysis"/>
        </Nav>
      </Router>
    );
  }
}

function SidebarLink({ label, to, activeOnlyWhenExact }) {
  return (
    <Route
      path={to}
      exact={activeOnlyWhenExact}
      children={({ match }) => (
        <Nav.Item bsPrefix="sidebaritem" id={label === "User" ? "user" : ""} className={match ? "active" : ""}>
          <Nav.Link bsPrefix="sidebarlink" href={to} eventKey={match ? "disabled" : ""}>
          { label }
          </Nav.Link>
        </Nav.Item>
      )}
    />
  );
}

export default SideBar;