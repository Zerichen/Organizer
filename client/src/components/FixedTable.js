import React, { Component } from 'react'
import '../css/TimeBlock.css';

class FixedTable extends Component{



    render(){
        return(
            <tr>
                    <td><p>{this.props.fName}</p>
                    </td>
                    
					<td><p>{this.props.fDate}</p>
                    </td>

                    <td><p>{this.props.fDuedate}</p>
                    </td>
					
                    
			</tr>
        )
    }


}

export default FixedTable;