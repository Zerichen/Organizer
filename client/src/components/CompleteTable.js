import React, { Component } from 'react'
import '../css/TimeBlock.css';
import FixedTable from './FixedTable';

class CompleteTable extends Component{
    

    render() {
		return (
		<table id = "tcomplete">
			<thead>  
				<tr>					
					<th>Dates: 3/10/2019</th> 
					<th>Estiamtion of time:</th>
					<th>Due Date:</th>
					<p id="demo"></p>

				</tr>
			</thead>

			<tbody>
                {this.props.fixedList.map((item,i)=>
				<FixedTable 
                    fName={item.taskName}
					fDate={item.taskDate}
					fDuedate={item.taskDuedate}
                    key={i}
                />
				)}
				
			</tbody>
			<script>
			
			</script>
			
		</table>

		)
	}
}

export default CompleteTable;