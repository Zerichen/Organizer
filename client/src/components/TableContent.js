import React, { Component } from 'react'
import { Row, Col, Button } from 'react-bootstrap';
import Change from './Change';
import "react-icons/io";
import '../css/TimeBlock.css';

//This a known issue 
//https://github.com/facebook/draft-js/issues/53

console.error = (function() {
  var error = console.error;
  return function(exception) {
    if ((exception + '').indexOf('Warning: A component is `contentEditable`') !== 0) {
      error.apply(console, arguments)
    }
  }
})()

class TableContent extends Component {
  constructor(){
    super();
    this.state = {
            
    };
  }
   
  // if the button edit is pressed, then the user may edit the name, estimated time, and due date
  handleEdit = () =>{
    this.refs.Name.contentEditable = 'true';
    this.refs.Estimated.contentEditable = 'true';
    this.refs.Duedate.contentEditable = 'true';
  }
    
  // if the mouse is pressed not in the editable content of name, then the user may no longer edit the value
  handleNameBlur = () =>{
    this.refs.Name.contentEditable = 'false';
    var name = this.refs.Name.innerHTML;
    this.props.handleNameBlur(name);
  }
    
  // if the mouse is pressed not in the editable content of estimated time, then the user may no longer edit the value
  handleEstimatedBlur = () =>{
    this.refs.Estimated.contentEditable = 'false';
    var estimated = this.refs.Estimated.innerHTML;
    this.props.handleEstimatedBlur(estimated);
  }
    
  // if the mouse is pressed not in the editable content of due date, then the user may no longer edit the value
  handleDuedateBlur = () =>{
    this.refs.Duedate.contentEditable = 'false';
    var subject = this.refs.Duedate.innerHTML;
    this.props.handleDuedateBlur(subject);
  }

  // onBlur is used to save the content of saving new data. 
  // When the mouse is no longer pointing at the value input
  // in this rendering case, it would be outputed value. 
  render() {
    return (
      <Row>
          <Col>
            <p contentEditable="false" ref='Subject' onBlur={this.handleNameBlur}>{this.props.subject}</p>
          </Col>
          <Col>
            <p contentEditable="false" ref='Name' onBlur={this.handleEstimatedBlur}> {this.props.name}</p>
          </Col>
          <Col>
            <p contentEditable="false" ref='Estimated' onBlur={this.handleDuedateBlur}> {this.props.estimated} Hours</p>
          </Col>
          <Col>
            <Button variant="light" size="sm" onClick={this.props.handleDone}>
              <ion-icon name="arrow-dropright-circle"></ion-icon>
            </Button> 
            <Button variant="light" size="sm" onClick={this.props.handleDone}>
              <ion-icon name="checkmark-circle-outline"></ion-icon>
            </Button> 
            <Change task={this.props.task}/>
            <Button variant="light" size="sm" onClick={this.props.handleDelete}>
              <ion-icon name="trash"></ion-icon>
            </Button>
          </Col>
      </Row>
    )
  }

}

export default TableContent;