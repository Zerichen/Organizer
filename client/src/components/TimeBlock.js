import React, { Component } from 'react';
import { Card, Row, Col } from 'react-bootstrap';
import TableContent from './TableContent';
import Change from './Change';
import '../css/TimeBlock.css';

class TimeBlock extends Component {
  constructor(props){
		super();
		this.state={
			date: props.date,
			tasks: props.tasks
		};
	}

	// handle deletion of an existing event
  handleDelete = (i) =>{
		this.state.tasks.splice(i,1);
		this.setState({
			tasks:this.state.tasks
		});
  }

  handleAddEvent = () =>{
		fetch('/schedules', { method: 'POST' })
		.then(res => res.json())
		.then(json => {
			const data = {};
			data.push(json);
			this.setState({
				schedules: data
			});
		});
  }

	// when done, the event would be put into the completed event box
  handleDone = (i) => {
		var taski = this.state.tasks[i];
		var newtasks = this.state.tasks;
		newtasks.splice(i,1);
		this.setState({
			tasks:newtasks
		});
    this.props.handleDone(taski);
	}

  render() {
		return (
			<Card id="cardlook" className="text-left">
				<Card.Header as="h5">
					{this.state.date}
				</Card.Header>

				<Card.Body>
					<Card.Title>
						<Row>
							<Col>Subject</Col>
							<Col>Required</Col>
							<Col>Estimated Time</Col>
							<Col> 
								<Change task={null}/>
							</Col>
						</Row>
					</Card.Title>
					{this.state.tasks.map((task,i)=>
						<TableContent 
							task={task}
							estimated={task.estimated}
							name={task.name}
							subject={task.class}
							key={i}
							handleDelete={this.handleDelete.bind(this,i)}
							handleDone={this.handleDone.bind(this,i)}
						/>
					)}
				</Card.Body>

			</Card>
		)
  }
}

export default TimeBlock;