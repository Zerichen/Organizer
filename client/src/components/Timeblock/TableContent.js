import React, { Component } from 'react'
import { Row, Col, Button } from 'react-bootstrap';
import Change from './Change';
import '../../css/TimeBlock.css';

class TableContent extends Component {
  constructor(props){
    super(props);

    this.handleUpdate = this.handleUpdate.bind(this);
    this.handleSave = this.handleSave.bind(this);

    this.state = {
      task: props.task,
      isStarted: props.task.hasOwnProperty('starttime') ? true : false,
      isDone: props.task.state
    };
  }

  handleStart = () => {
    this.setState({isStarted: true}); // to delete after connect to db
    fetch('/task', {
      method: 'UPDATE',
      body: JSON.stringify({taskid: 'somestring'}), // set taskid
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      }
    }).then(res => res.json())
		.then((data) => console.log(data))
    .catch((err)=>console.log(err))
  }

  handleUpdate(updateInfo) {
    this.setState({
      task: updateInfo
    });

    this.props.handleUpdate(updateInfo);
  }

  handleSave(saveInfo) {
    this.props.handleSave(saveInfo);
  }
  
  handleDone = () => {
    this.setState({isDone: true});
    fetch('/done', {
      method: 'UPDATE',
      body: JSON.stringify({taskid: 'somestring'}), // set taskid
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      }
    }).then(res => res.json())
		.then((data) => console.log(data))
    .catch((err)=>console.log(err))
  }

  render() {
    let duetime = this.props.task.duetime.slice(11,16);
    const h = parseInt(duetime.slice(0,2));
    let indicator = 'AM';
    if (h >= 12) {
      if (h !== 12)
        duetime = (h-12).toString() + ':' + duetime.slice(3,5);
      indicator = 'PM';
    }
    return (
      <Row 
        bsPrefix={this.state.isDone ? "row done" : "row"} 
        style={{ textAlign: 'center' }}
      >
          <Col>
            <p ref='Subject'>{this.state.task.class}</p>
          </Col>
          <Col>
            <p ref='Name'> {this.state.task.name}</p>
          </Col>
          <Col>
            <p ref='Duetime'> { duetime } { indicator } </p>
          </Col>
          <Col>
            <p ref='Estimated'> {this.state.task.estimated} Hours</p>
          </Col>
          <Col>
            {this.state.task.tag.map((tag, i) => 
              <font 
                key={i}
                ref='Tag' 
                color={tag.color}
              > 
                {tag.name} &nbsp;
              </font>
            )}
          </Col>
          <Col>
            <Button 
              variant="light" 
              size="sm" 
              disabled={this.state.isStarted || this.state.isDone} 
              onClick={this.handleStart}
            >
              <ion-icon name="arrow-dropright-circle"></ion-icon>
            </Button> 
            <Button 
              variant="light" 
              size="sm" 
              disabled={this.state.isDone} 
              onClick={this.handleDone}
            >
              <ion-icon name="checkmark-circle-outline"></ion-icon>
            </Button> 
            <Change 
              disabled={this.state.isDone} 
              task={this.state.task}
              handleUpdate={this.handleUpdate}
              handleSave={this.handleSave}
            />
            <Button 
              variant="light" 
              size="sm" 
              disabled={this.state.isDone} 
              onClick={this.props.handleDelete}
            >
              <ion-icon name="trash"></ion-icon>
            </Button>
          </Col>
      </Row>
    )
  }

}

export default TableContent;