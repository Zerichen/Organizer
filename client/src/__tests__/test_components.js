import React from 'react';
import ReactDOM from 'react-dom';
import Events from '../components/Timeline/Events.jsx';
import Sidebar from '../components/Sidebar.js';
import renderer from 'react-test-renderer';
import CompleteTable from '../components/CompleteTable.js';
import TimeBlock from '../components/TimeBlock.js';
import TableContent from '../components/TableContent.js';
import FixedTable from '../components/FixedTable.js';


it('renders without crashing', () => {
    const div = document.createElement('div');
    const tbody = document.createElement('tbody');
    ReactDOM.render(<Sidebar/ >, div);
});

it('timeline renders correctly', () => {
    const testarray = {'March 23':["1", "2"], "April 2":["3"], "May 19":["4"]};
    const ele = renderer.create(<Events eventArray={testarray}/>).toJSON();
    expect(ele).toMatchSnapshot();
});
