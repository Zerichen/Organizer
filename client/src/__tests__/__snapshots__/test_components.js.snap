// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timeline renders correctly 1`] = `
<span>
  <i
    className="arrow left"
  />
  <ol
    className="eventsList"
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <ul
      className="future"
      data-radium={true}
      id="March 23"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "borderRadius": "10%",
          "color": "#0f0f0f",
          "height": "30px",
          "left": "350px",
          "position": "absolute",
          "textAlign": "center",
          "top": 80,
          "width": "100px",
        }
      }
    >
      March 23
      <li
        className="Hwitem"
        data-radium={true}
        style={
          Object {
            "display": "none",
          }
        }
      >
        1
      </li>
      <li
        className="Hwitem"
        data-radium={true}
        style={
          Object {
            "display": "none",
          }
        }
      >
        2
      </li>
    </ul>
    <ul
      className="future"
      data-radium={true}
      id="April 2"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "borderRadius": "10%",
          "color": "#0f0f0f",
          "height": "30px",
          "left": "500px",
          "position": "absolute",
          "textAlign": "center",
          "top": 80,
          "width": "100px",
        }
      }
    >
      April 2
      <li
        className="Hwitem"
        data-radium={true}
        style={
          Object {
            "display": "none",
          }
        }
      >
        3
      </li>
    </ul>
    <ul
      className="future"
      data-radium={true}
      id="May 19"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "borderRadius": "10%",
          "color": "#0f0f0f",
          "height": "30px",
          "left": "650px",
          "position": "absolute",
          "textAlign": "center",
          "top": 80,
          "width": "100px",
        }
      }
    >
      May 19
      <li
        className="Hwitem"
        data-radium={true}
        style={
          Object {
            "display": "none",
          }
        }
      >
        4
      </li>
    </ul>
  </ol>
  <i
    className="arrow right"
  />
</span>
`;
